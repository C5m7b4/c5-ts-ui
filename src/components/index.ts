import {
  Button,
  NormalButtonType,
  NormalButtonSize,
  ButtonProps,
} from './Button';

export { Button, NormalButtonType, NormalButtonSize, ButtonProps };
