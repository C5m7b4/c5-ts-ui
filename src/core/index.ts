import Backdrop, { BackdropProps } from './Backdrop';
import Modal, { ModalProps } from './Modal';
import ModalWrapper, { ModalWrapperProps } from './ModalWrapper';

export {
  Backdrop,
  BackdropProps,
  Modal,
  ModalProps,
  ModalWrapper,
  ModalWrapperProps,
};
