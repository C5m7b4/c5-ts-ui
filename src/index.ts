import { Add } from './math';

export { Add };
