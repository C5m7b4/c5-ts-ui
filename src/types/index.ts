export type Theme = 'light' | 'dark' | 'colored';

export type Parsers =
  | 'babel'
  | 'css'
  | 'json'
  | 'flow'
  | 'babel-flow'
  | 'babel-ts'
  | 'typescript'
  | 'json'
  | 'markdown'
  | 'html';
