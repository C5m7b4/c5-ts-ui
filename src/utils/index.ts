export * from './colors';
export * from './reserved';
export * from './fns';
