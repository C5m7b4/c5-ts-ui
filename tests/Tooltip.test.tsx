import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { Tooltip } from '../src';

describe('Tooltip tests', () => {
  test('should render correctly', () => {
    const { container } = render(
      <Tooltip position="top" message="tooltip">
        <h2>Hello</h2>
      </Tooltip>
    );
    expect(container).toMatchSnapshot();
  });

  test('should handle some style', () => {
    const { container } = render(
      <Tooltip
        position="top"
        message="tooltip"
        style={{ backgroundColor: 'red' }}
      >
        <h2>Hello</h2>
      </Tooltip>
    );
    expect(container).toMatchSnapshot();
  });

  test('should handle mouseOver', async () => {
    const { container } = render(
      <Tooltip position="top" message="tooltip">
        <div>Hello</div>
      </Tooltip>
    );

    const header = container.querySelector(
      '.mcl-tooltip-trigger'
    ) as HTMLDivElement;
    fireEvent.mouseOver(header);
    fireEvent.mouseLeave(header);
    fireEvent.mouseOut(header);
  });

  test('should handle dark mode', () => {
    const { container } = render(
      <Tooltip position="bottom" message="tooltip" theme="dark">
        <button>Hello</button>
      </Tooltip>
    );
    expect(container).toMatchSnapshot();
  });

  test('should render without a position', () => {
    const { container } = render(
      <Tooltip message="tooltip" theme="light">
        <div>Hellos</div>
      </Tooltip>
    );
    expect(container).toMatchSnapshot();
  });
});
