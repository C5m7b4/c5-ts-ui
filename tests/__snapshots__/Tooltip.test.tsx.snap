// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip tests should handle dark mode 1`] = `
<div>
  <span
    class="mcl-tooltip-wrapper"
  >
    <span
      class="mcl-tooltip-trigger"
    >
      <button>
        Hello
      </button>
    </span>
  </span>
</div>
`;

exports[`Tooltip tests should handle some style 1`] = `
<div>
  <span
    class="mcl-tooltip-wrapper"
  >
    <span
      class="mcl-tooltip-trigger"
    >
      <h2>
        Hello
      </h2>
    </span>
  </span>
</div>
`;

exports[`Tooltip tests should render correctly 1`] = `
<div>
  <span
    class="mcl-tooltip-wrapper"
  >
    <span
      class="mcl-tooltip-trigger"
    >
      <h2>
        Hello
      </h2>
    </span>
  </span>
</div>
`;

exports[`Tooltip tests should render without a position 1`] = `
<div>
  <span
    class="mcl-tooltip-wrapper"
  >
    <span
      class="mcl-tooltip-trigger"
    >
      <div>
        Hellos
      </div>
    </span>
  </span>
</div>
`;
